### PR TITLE
chore: add acceptance tests and migration tests to report in merge queue

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s):

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

## Further comments
Follow up of https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1588

### CODE REVIEW

**Review by gssbzn — APPROVED** [review]

File: .github/workflows/acceptance-tests.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ on:
         type: string
   pull_request: # you can run a specic job in your PR using GitHub labels
     types: [ labeled ]
+  merge_group: # needed for merge queue
```
Comment (AgustinBettati) [inline .github/workflows/acceptance-tests.yml:14]: Not strictly related to these changes, but I see from latest build (https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/6745326427) and it seems merge_group does not work well with `dorny/paths-filter@v2`. I think we had a similar issue in cloud formation.

Comment (andreaangiolillo) [reply .github/workflows/acceptance-tests.yml:14]: the problem is that code health is calling the acceptance test action which fails in the merge queue 🤔

Comment (andreaangiolillo) [reply .github/workflows/acceptance-tests.yml:14]: found a possible fix in https://github.com/PyO3/pyo3/pull/3212

Comment (andreaangiolillo) [reply .github/workflows/acceptance-tests.yml:14]: checking the merge queue

Comment (AgustinBettati) [reply .github/workflows/acceptance-tests.yml:14]: interesting fix, nice find

**Pull request merged**